tests/foreman/ui/test_host.py

```diff
@@ -158,6 +158,10 @@ def module_global_params():
 
 @pytest.fixture(scope='module')
 def module_host_template(module_org, module_loc):
+    proxy = entities.SmartProxy().search(
+        query={u'search': u'name={0}'.format(settings.server.hostname)})[0].read()
```

Comment:
do we require to use read() explicitly or the only search will take care?

Reply:
yeah it requires to upgrade the location of proxy host:

here an example of diff with and without of read():

```
(Pdb) proxy = entities.SmartProxy().search(query={u'search': u'name={0}'.format(settings.server.hostname)})[0]
(Pdb) dir(proxy)
['__add__', '__class__', '__contains__', '__delattr__', '__delitem__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__iadd__', '__imul__', '__init__', '__init_subclass__', '__iter__', '__le__', '__len__', '__lt__', '__mul__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__reversed__', '__rmul__', '__setattr__', '__setitem__', '__sizeof__', '__str__', '__subclasshook__', 'append', 'clear', 'copy', 'count', 'extend', 'index', 'insert', 'pop', 'remove', 'reverse', 'sort']




(Pdb) proxy = entities.SmartProxy().search(query={u'search': u'name={0}'.format(settings.server.hostname)})[0].read()
(Pdb) dir(proxy)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_fields', '_meta', '_server_config', 'compare', 'create', 'create_json', 'create_missing', 'create_payload', 'create_raw', 'delete', 'delete_raw', 'get_fields', 'get_values', 'id', 'import_puppetclasses', 'location', 'name', 'organization', 'path', 'read', 'read_json', 'read_raw', 'refresh', 'search', 'search_filter', 'search_json', 'search_normalize', 'search_payload', 'search_raw', 'to_json', 'to_json_dict', 'update', 'update_json', 'update_payload', 'update_raw', 'url']
(Pdb) 
```
```

Reply:
make sense, I thought that search()[0] return that object. I'm good with it.

Reply:
Just to let you know, `search()` **can** return full object, but you have to use `filters` arg:
```python
entities.SmartProxy().search(filters={'name': '{0}'.format(settings.server.hostname)})[0]
```

This way nailgun will get full details of all Smart Proxies and then filter them locally. This should be always slower than search and then read, but allows for some more concise code.

Reply:
@mirzal thanks for that